%% LyX 1.3 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
\documentclass[english, 12pt]{article}
\usepackage{times}
%\usepackage{algorithm2e}
\usepackage{url}
\usepackage{bbm}
\usepackage[T1]{fontenc}
\usepackage[latin1]{inputenc}
\usepackage{geometry}
\geometry{verbose,letterpaper,tmargin=2cm,bmargin=2cm,lmargin=1.5cm,rmargin=1.5cm}
\usepackage{rotating}
\usepackage{color}
\usepackage{graphicx}
\usepackage{subcaption}
\usepackage{amsmath, amsthm, amssymb}
\usepackage{setspace}
\usepackage{lineno}
\usepackage{hyperref}
\usepackage{bbm}
\usepackage{makecell}
\usepackage{placeins}

\renewcommand{\arraystretch}{1.2}

%\usepackage{xr}
%\externaldocument{SCT-supp}

%\linenumbers
%\doublespacing
\onehalfspacing
%\usepackage[authoryear]{natbib}
\usepackage{natbib} \bibpunct{(}{)}{;}{author-year}{}{,}

%Pour les rajouts
\usepackage{color}
\definecolor{trustcolor}{rgb}{0,0,1}

\usepackage{dsfont}
\usepackage[warn]{textcomp}
\usepackage{adjustbox}
\usepackage{multirow}
\usepackage{graphicx}
\graphicspath{{../figures/}}
\DeclareMathOperator*{\argmin}{\arg\!\min}

\let\tabbeg\tabular
\let\tabend\endtabular
\renewenvironment{tabular}{\begin{adjustbox}{max width=0.95\textwidth}\tabbeg}{\tabend\end{adjustbox}}

\makeatletter

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% LyX specific LaTeX commands.
%% Bold symbol macro for standard LaTeX users
%\newcommand{\boldsymbol}[1]{\mbox{\boldmath $#1$}}

%% Because html converters don't know tabularnewline
\providecommand{\tabularnewline}{\\}

\usepackage{babel}
\makeatother


\begin{document}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\clearpage

\section*{Supplementary Materials}

\renewcommand{\thefigure}{S\arabic{figure}}
\setcounter{figure}{0}
\renewcommand{\thetable}{S\arabic{table}}
\setcounter{table}{0}
\renewcommand{\theequation}{S\arabic{equation}}
\setcounter{equation}{0}
\renewcommand{\thesection}{S\arabic{section}}
\setcounter{section}{1}

\subsection*{Definitions $\dag$ and methods}

\begin{itemize}
	
	\item The {\bf 1000 Genomes Project (1000G)} data is composed of approximately 100 individuals for each of 26 populations worldwide (described at \url{https://www.internationalgenome.org/category/population/}), including 7 African (AFR), 5 East Asian (EAS), 5 South Asian (SAS), 5 European (EUR) and 4 admixed American (AMR) populations.
	Here we used the transformed data in PLINK format provided in \cite{prive2020efficient}.
	
	\item The $\boldsymbol{F_{ST}}$ measures the relative amount of genetic variance between populations compared to the total genetic variance within these populations \cite[]{wright1965interpretation}.
	We use the weighted average formula proposed in \cite{weir1984estimating}, which we now implement in our package bigsnpr \cite[]{prive2017efficient}.
	
	\item The {\bf Principal Component (PC) scores} are defined as $U \Delta$, where $U \Delta V^T$ is the singular value decomposition of the (scaled) genotype matrix \cite[]{prive2020efficient}. They are usually truncated, e.g.\ corresponding to the first 20 principal dimensions only. 
	
	\item The {\bf Bhattacharyya distance} between two multivariate normal distributions $\mathcal{N}(\boldsymbol\mu_1,\,\boldsymbol\Sigma_1)$ and $\mathcal{N}(\boldsymbol\mu_2,\,\boldsymbol\Sigma_2)$ is defined as
	$D_B={1\over 8}(\boldsymbol\mu_2-\boldsymbol\mu_1)^T \boldsymbol\Sigma^{-1}(\boldsymbol\mu_2-\boldsymbol\mu_1)+{1\over 2}\log \,\left({|\boldsymbol\Sigma| \over \sqrt{|\boldsymbol\Sigma_1| \, |\boldsymbol\Sigma_2|} }\right)$,
	where $\boldsymbol\Sigma={\boldsymbol\Sigma_1+\boldsymbol\Sigma_2 \over 2}$ and $|M|$ is the absolute value of the determinant of matrix $M$ \cite[]{bhattacharyya1943measure,fukunaga1990introduction}. 
	The mean and covariance parameters for each population are computed using the robust location and covariance parameters as proposed in \cite{prive2020efficient}.
	
	\item The {\bf geometric median} of points is the point that minimizes the sum of all Euclidean distances to these points. We now implement this as function \texttt{geometric\_median} in our R package bigutilsr.
	
	\item The {\bf UK Biobank} is a large cohort of half a million individuals from the UK, for which we have access to both genotypes and multiple phenotypes (\url{https://www.ukbiobank.ac.uk/}).
	We apply some quality control filters to the genotyped data; we remove individuals with more than 10\% missing values, variants with more than 1\% missing values, variants having a minor allele frequency < 0.01, variants with P-value of the Hardy-Weinberg exact test < $10^{-50}$, and non-autosomal variants. 
	This results in 488,371 individuals and 504,139 genetic variants.
	
\end{itemize}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\clearpage

\subsection*{Measures of genetic dissimilarity between populations}

\input{../tables/fst-values-AFR}
\input{../tables/fst-values-AMR}
\input{../tables/fst-values-EAS}
\input{../tables/fst-values-EUR}
\input{../tables/fst-values-SAS}

\FloatBarrier

\begin{figure}[h]
\centerline{\includegraphics[width=0.8\textwidth]{heatmap-Fst-1000G}}
\caption{Heatmap with clustering based on the $F_{ST}$ between pairs of the 26 1000G populations. Corresponding values are reported in tables \ref{tab:fst-AFR}-\ref{tab:fst-SAS}. \label{fig:heatmap0}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.8\textwidth]{heatmap-bhattacharyya-1000G}}
	\caption{Heatmap with clustering based on the Bhattacharyya distances between pairs of the 26 1000G populations. \label{fig:heatmap1}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.7\textwidth]{compare-Bhattacharyya-to-Fst}}
	\caption{Comparing $F_{ST}$ to the Bhattacharyya distance on the PCA space between pairs of the 26 1000G populations. \label{fig:compare-dist1}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.8\textwidth]{heatmap-centers-1000G}}
	\caption{Heatmap with clustering based on the Euclidean distances between centers of pairs of the 26 1000G populations. \label{fig:heatmap2}}
\end{figure}

\begin{figure}[h]
\centerline{\includegraphics[width=0.95\textwidth]{compare-Euclidean-to-Fst2}}
\caption{Comparing $F_{ST}$ to the squared Euclidean distances on the PCA space between centers of pairs of the 26 1000G populations. Distances are computed using different numbers of Principal Components (PCs). \label{fig:compare-distK}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.8\textwidth]{heatmap-closest-1000G}}
	\caption{Heatmap with clustering based on the shortest distances between individuals in pairs of the 26 1000G populations. \label{fig:heatmap3}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.7\textwidth]{compare-closest-to-Fst}}
	\caption{Comparing $F_{ST}$ to the shortest distances between individuals in pairs of the 26 1000G populations. \label{fig:compare-dist3}}
\end{figure}

\FloatBarrier

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{PCA-based ancestry inference}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.95\textwidth]{proj-UKBB-to-1000G}}
	\caption{First 18 PC scores of the 1000G data (in black), onto which the UK Biobank data has been projected (in red). \label{fig:proj-UKBB}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.95\textwidth]{UKBB-mismatch}}
	\caption{PC scores (computed in the UK Biobank) colored by self-reported ancestry. On the left, these are 50,000 random individuals. On the right, these are the 47 individuals with some discrepancy between their self-reported-ancestry and our ancestry estimation. \label{fig:mismatch}}
\end{figure}

\input{../tables/matched-pop-KNN}
\input{../tables/infer-UKBB-pop}
\input{../tables/POPRES-matching}


\FloatBarrier

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{PCA-based ancestry grouping}

\input{../tables/ancestry-groups}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.9\textwidth]{heatmap-country-UKBB}}
	\caption{Heatmap with clustering based on the distances in the PCA space between centers of pairs of the countries of birth in the UK Biobank. \label{fig:heatmap-country}}
\end{figure}

\FloatBarrier

\input{../tables/country-groups}

\begin{figure}[htb]
	\centerline{\includegraphics[width=0.95\textwidth]{UKBB-matched-country}}
	\caption{The first eight PC scores computed from the UK Biobank (Field 22009) colored by the homogeneous ancestry group we infer for these individuals. \label{fig:grouping2}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.8\textwidth]{hist-dist-overall-center}}
	\caption{Histogram of (log) squared distances from the UK Biobank PC scores to the geometric median of the all UKBB individuals. Here we use a threshold at 7, based on visual inspection. Alternatively, a more stringent threshold at 6 could also be used. \label{fig:hist3}}
\end{figure}

\FloatBarrier

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\clearpage

\bibliographystyle{natbib}
\bibliography{refs}



\end{document}
