%% LyX 1.3 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
\documentclass[english, 12pt]{article}
\usepackage{times}
%\usepackage{algorithm2e}
\usepackage{url}
\usepackage{bbm}
\usepackage[T1]{fontenc}
\usepackage[latin1]{inputenc}
\usepackage{geometry}
\geometry{verbose,letterpaper,tmargin=2cm,bmargin=2cm,lmargin=1.5cm,rmargin=1.5cm}
\usepackage{rotating}
\usepackage{color}
\usepackage{graphicx}
\usepackage{subcaption}
\usepackage{amsmath, amsthm, amssymb}
\usepackage{setspace}
\usepackage{lineno}
\usepackage{hyperref}
\usepackage{bbm}
\usepackage{makecell}
\usepackage{placeins}

\renewcommand{\arraystretch}{1.2}

%\usepackage{xr}
%\externaldocument{SCT-supp}

%\linenumbers
%\doublespacing
\onehalfspacing
%\usepackage[authoryear]{natbib}
\usepackage{natbib} \bibpunct{(}{)}{;}{author-year}{}{,}

%Pour les rajouts
\usepackage{color}
\definecolor{trustcolor}{rgb}{0,0,1}

\usepackage{dsfont}
\usepackage[warn]{textcomp}
\usepackage{adjustbox}
\usepackage{multirow}
\usepackage{graphicx}
\graphicspath{{../figures/}}
\DeclareMathOperator*{\argmin}{\arg\!\min}

\let\tabbeg\tabular
\let\tabend\endtabular
\renewenvironment{tabular}{\begin{adjustbox}{max width=0.9\textwidth}\tabbeg}{\tabend\end{adjustbox}}

\makeatletter

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% LyX specific LaTeX commands.
%% Bold symbol macro for standard LaTeX users
%\newcommand{\boldsymbol}[1]{\mbox{\boldmath $#1$}}

%% Because html converters don't know tabularnewline
\providecommand{\tabularnewline}{\\}

\usepackage{babel}
\makeatother


\begin{document}


\title{Ancestry inference and grouping\\from principal component analysis of genetic data
}
\author{Florian Priv\'e$^{\text{1,}*}$}

\date{~ }
\maketitle

\noindent$^{\text{\sf 1}}$National Centre for Register-Based Research, Aarhus University, Aarhus, 8210, Denmark. \\
\noindent$^\ast$To whom correspondence should be addressed.\\

\noindent Contact:
\begin{itemize}
\item \url{florian.prive.21@gmail.com}
\end{itemize}

\vspace*{5em}

\abstract{
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\newpage

$^\dag$ Further defined in supplementary section ``Defintions''.

\subsection*{Introduction}

Why do we need to group individuals in ancestry groups?

There is no consensus on the method to use to infer population structure, but particularly on grouping individuals.

Why it makes sense to use PCA-based distances -> correction for population structure, and geographical distance. + Fast

Global ancestry only

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{Measures of genetic dissimilarity between populations}

We first compare four measures of genetic dissimilarity using populations of the 1000 genomes as example \cite[]{10002015global}. 
The $F_{ST}$$^\dag$ is an ubiquitous measure of genetic dissimilarity between populations and the first measure we use in this comparison.
We report $F_{ST}$ between the 26 1000G populations in tables S1-S5.
The other three measures compared are distances applied to the PC scores$^\dag$ of the genetic data: 1) the Bhattacharyya distance$^\dag$; 2) the distance between the centers (geometric median$^\dag$) of the two populations; 3) the shorter distance between pairs of PC scores from the two populations.
The distance between population centers seems to be an appropriate PCA-based distance as this (squared) distance is approximately proportional to the $F_{ST}$ (Figure \ref{fig:compare-dist}) and provides an appropriate clustering of populations (Figure \ref{fig:heatmap2}).
In contrast, the two other Bhattacharyya and shortest distances do not provide as satisfactory results (Figures \ref{fig:heatmap1} and \ref{fig:heatmap3}). For example, using , African Caribbeans in Barbados (ACB) and Americans of African Ancestry in SW USA (ASW) are close to Northern European populations (FIN, CEU and GBR) when using the Bhattacharyya distance.
Using the shortest distance between pairs of individuals in two different populations is very sensitive to outliers being far from the other individuals in the population.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{PCA-based ancestry inference}

We project the dataset of interest (either UKBB or POPRES) onto the PCA space of the 1000G data using the fast tools developed in \cite{prive2020efficient}.
We recall that this uses a correction for shrinkage in projected PC scores, which has been shown to be particularly important when using PC scores for ancestry estimation \cite[]{zhang2020fast}. 
Based on the results of the previous section, we propose to assign individual ancestry to one of the 26 1000G populations based on the minimum distance to their respective center (geometric median$^\dag$). 
Since we showed previously that (squared) distances in the PCA space are proportional to $F_{ST}$, we can set a threshold on these distances that would correspond approximately to an $F_{ST}$ of e.g. 0.002. This threshold is close to the dissimilarity between Spanish and Italian people ($F_{ST}$(IBS, TSI) of 0.0015).
When an individual is not close enough to any of the 26 1000G populations, we leave its ancestry inference as unknown.

We first perform ancestry estimation for the individuals in the UK Biobank$^\dag$.
These individuals seem to originate from all over the world when we project onto the PCA space of the 1000G (Figure \ref{fig:proj-UKBB}). 
Self-reported ancestry (Field 21000) is available for almost all individuals, with only 1.6\% with unknown or mixed ancestry.
Based on the threshold defined before, we could not infer ancestry for 4.6\% of the individuals.
More precisely, among ``British'', ``Irish'' and ``White'' ancestries, this represented respectively 2.2\%, 3.3\% and 7.9\% (Tables \ref{tab:infer-UKBB-superpop} and \ref{tab:ancestry-fine-pred}). This also represented 3.3\% for ``Chinese'', 13.8\% for ``Indian'' and 17.8\% for ``African'' ancestries. 
Finally, mixed ancestries were particularly difficult to match to any of the 1000G populations, e.g. 97.3\% unmatched within `White and Black African` and 93.0\% within `White and Asian`.
Only 47 individuals were wrongly classified in ``super'' population of the 1000G; e.g.\ six ``British'' were classified as South Asians, one ``Chinese'' as European and 25 ``Caribbean'' as South Asian by our method (Table \ref{tab:infer-UKBB-superpop}). 
However, when comparing the location of these individuals on the PCA space computed in the UK Biobank \cite[]{bycroft2017genome} to the rest of individuals, it seemed more probable that the self-reported ancestry is wrong and not our ancestry estimate (Figure \ref{fig:mismatch}).

We also test the approach proposed in \cite{zhang2020fast} which consists of finding the 20 nearest neighbors in 1000G and computing the frequency of (super) population membership, weighted by the inverse distance to these 20 closest 1000G individuals. When this probability is less than 0.875, they leave the ancestry as unknown, aiming at discarding admixed individuals.
Less than 0.5\% could not be matched by their method (Table \ref{tab:ancestry-pred-kNN}).
Of note, they could match much more admixed individuals, whereas they set a high probably aiming at discarding such admixed individuals. 
Morever, there are many more discrepancies between their method and the self-reported ancestry in the UK Biobank (Table \ref{tab:ancestry-pred-kNN}) compared to the previous results with our method (Table  \ref{tab:infer-UKBB-superpop}).
Finally, our method is able to accurately differentiate between sub-continental populations such as differentiating between Pakistani, Bangladeshi and Chinese people (Table \ref{tab:ancestry-fine-pred}).
We also applied our ancestry detection technique to the European individuals of the POPRES data \cite[]{nelson2008population}. 
Only 16 out of the 1385 individuals (1.2\%) could not be matched, of which 11 were from East or South-East Europe (Table \ref{tab:ancestry-pred-popres}).
Note that all individuals that we could match were identified as of European ancestry. 
We could also identify accurately sub-regions of Europe; e.g.\ 261 out of 264 Spanish and Portugese individuals were identified as ``Iberian Population in Spain'' (EUR\_IBS, Table \ref{tab:ancestry-pred-popres}).

\begin{table}[htb]
	\centering
	\caption{Self-reported ancestry (left) of UKBB individuals and their matching to 1000G continental populations (top) by our method. See the description of 1000G populations at \url{https://www.internationalgenome.org/category/population/}.} 
	\label{tab:infer-UKBB-superpop}
	\begin{tabular}{|l|c|c|c|c|c|c|}
		\hline
		& AFR & AMR & EAS & EUR & SAS & Not matched \\ 
		\hline
		British & 2 &  & 1 & 421457 & 6 & 9548 \\ 
		Irish &  &  &  & 12328 &  & 425 \\ 
		White & 1 & 1 & 1 & 499 &  & 43 \\ 
		Other White &  & 40 &  & 11334 & 1 & 4440 \\ 
		\hline
		Indian &  &  &  & 5 & 4922 & 789 \\ 
		Pakistani &  &  &  &  & 1421 & 327 \\ 
		Bangladeshi &  &  &  &  & 217 & 4 \\ 
		Chinese &  &  & 1453 & 1 &  & 50 \\ 
		Other Asian & 1 &  & 279 &  & 939 & 528 \\ 
		\hline
		Caribbean & 3848 &  &  &  & 25 & 424 \\ 
		African & 2633 &  &  & 1 &  & 570 \\ 
		Other Black & 74 &  &  &  & 2 & 42 \\ 
		\hline
		Asian or Asian British &  &  & 2 &  & 20 & 20 \\ 
		Black or Black British & 20 &  &  & 2 &  & 4 \\ 
		White and Black Caribbean & 24 & 1 &  & 8 & 1 & 563 \\ 
		White and Black African & 5 &  &  & 6 &  & 391 \\ 
		White and Asian &  & 1 & 2 & 27 & 26 & 746 \\ 
		\hline
		Unknown  & 835 & 173 & 576 & 2296 & 633 & 3307 \\ 
		\hline
	\end{tabular}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{PCA-based ancestry grouping}

Finally, we show how to use our ancestry inference method for grouping genetically homogeneous individuals.
We identify 3 possible ways of doing this.
The first approach is to simply match identify individuals that are closed enough to one or more of the 1000G populations, as described in this paper.
We can e.g. define a British ancestry cluster by identifying individuals close enough to the GBR 1000G population (Figure \ref{fig:hist1}).
Likewise, we can define ``AFR'', ``AMR'', ``EAS'' and ``SAS'' clusters of individuals who are close enough to any of the corresponding 1000G populations.
We can then verify that these clusters define distinct sets of homogeneous individuals (Figure \ref{fig:grouping}).
Another solution, which does not require projecting individuals to the 1000G, but does require computing PC scores in the dataset instead, would consist in using reported ancestries and internal PC scores.
Then, a British ancestry cluster could be defined by individuals close enough to the center of the people reported as ``British''  (Figure \ref{fig:hist2}).
We can also define an ``AFR2'' cluster if close enough to the center of either ``Caribbean'', ``African'' or ``Other Black'' individuals, ``EAS2'' if close to the ``Chinese'' cluster, and ``SAS2'' if close to either ``Indian'', ``Pakistani'' or ``Bangladeschi'' clusters.
This enables to capture a larger set of individuals who are close enough to British people (e.g. Irish people), while discarding individuals whose genetic ancestry is not really ``British''. 
A third solution, useful when we know that the data is composed of an overdominant ancestry is to keep individuals who are close enough to the center of all individuals (Figure \ref{fig:hist3}).
We recall the we are using the geometric median as such center, which is robust to outliers.
Overall, all these techniques approximately identify the same amount of people (Table \ref{tab:ancestry-groups}).

\begin{figure}[htb]
	\centerline{\includegraphics[width=0.95\textwidth]{UKBB-matched-ancestry}}
	\caption{ \label{fig:grouping}}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{Discussion}

Global measures vs relative measures.

Contrast with Mahalanobis distance proposed before.

Number of PCs -> 2 is not enough

\noindent[DISCUSS LACK OF SOME POPULATIONS IN 1000G]

\noindent[DISCUSS PC-BASED GENETIC DISTANCES BETWEEN POPULATIONS]

\noindent[IMPLICATIONS FOR PRS?]

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\clearpage

\section*{Software and code availability}

%R package pcadapt is available on CRAN.
%A tutorial on using pcadapt to detect local adaptation is available at \url{https://bcm-uga.github.io/pcadapt/articles/pcadapt.html}.
%The code used in this paper is available at \url{https://github.com/bcm-uga/pcadapt/tree/master/new-paper/code}.

\section*{Acknowledgements}

\section*{Funding}

F.P.\ is supported by the Danish National Research Foundation (Niels Bohr Professorship to John McGrath).

\section*{Declaration of Interests}

The authors declare no competing interests.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\newpage

\bibliographystyle{natbib}
\bibliography{refs}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Supplementary Materials}

\renewcommand{\thefigure}{S\arabic{figure}}
\setcounter{figure}{0}
\renewcommand{\thetable}{S\arabic{table}}
\setcounter{table}{0}
\renewcommand{\theequation}{S\arabic{equation}}
\setcounter{equation}{0}
\renewcommand{\thesection}{S\arabic{section}}
\setcounter{section}{1}

\subsection*{Definitions $\dag$ and methods}

[ALSO DESCRIBE 1000G DATA]

\begin{itemize}
	
	\item The $\boldsymbol{F_{ST}}$ measures the relative amount of genetic variance between populations compared to the total genetic variance within these populations \cite[]{wright1965interpretation}.
	We use the weighted average formula proposed in \cite{weir1984estimating}, which we now implement in our package bigsnpr \cite[]{prive2017efficient}.
	
	\item The {\bf Principal Component (PC) scores} are defined as $U \Delta$, where $U \Delta V^T$ is the singular value decomposition of the (scaled) genotype matrix. They are usually truncated, e.g.\ corresponding to the first 20 principal dimensions only. 
	
	\item The {\bf Bhattacharyya distance} between two multivariate normal distributions $\mathcal{N}(\boldsymbol\mu_1,\,\boldsymbol\Sigma_1)$ and $\mathcal{N}(\boldsymbol\mu_2,\,\boldsymbol\Sigma_2)$, which is defined as
	$D_B={1\over 8}(\boldsymbol\mu_2-\boldsymbol\mu_1)^T \boldsymbol\Sigma^{-1}(\boldsymbol\mu_2-\boldsymbol\mu_1)+{1\over 2}\log \,\left({|\boldsymbol\Sigma| \over \sqrt{|\boldsymbol\Sigma_1| \, |\boldsymbol\Sigma_2|} }\right)$,
	where $\boldsymbol\Sigma={\boldsymbol\Sigma_1+\boldsymbol\Sigma_2 \over 2}$ and $|M|$ is the absolute value of the determinant of matrix $M$ \cite[]{bhattacharyya1943measure,fukunaga1990introduction}. 
	The mean and covariance parameters for each population are computed using the robust location and covariance parameters as proposed in \cite{prive2020efficient}.
	
	\item The {\bf geometric median} of points is the point that minimizes the sum of all Euclidean distances to these points. We now implement this as function \texttt{geometric\_median} in our R package bigutilsr.
	
	\item The {\bf UK Biobank} is a large cohort of half a million individuals from the UK, for which we have access to both genotypes and multiple phenotypes (\url{https://www.ukbiobank.ac.uk/}).
	We apply some quality control filters to the genotyped data; we remove individuals with more than 10\% missing values, variants with more than 1\% missing values, variants having a minor allele frequency < 0.01, variants with P-value of the Hardy-Weinberg exact test < $10^{-50}$, and non-autosomal variants. 
	This results in 488,371 individuals and 504,139 genetic variants.
	
\end{itemize}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\clearpage

\subsection*{Location of 1000G populations}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.95\textwidth]{map-1000G}}
	\caption{ \label{fig:map}}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\clearpage

\subsection*{Measures of genetic dissimilarity between populations}

\input{../tables/fst-values-AFR}
\input{../tables/fst-values-AMR}
\input{../tables/fst-values-EAS}
\input{../tables/fst-values-EUR}
\input{../tables/fst-values-SAS}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.8\textwidth]{compare-dist-to-Fst}}
	\caption{Comparing Fst to different distances in the PCA space between pairs of the 26 1000G populations. \label{fig:compare-dist}}
\end{figure}

\begin{figure}[h]
\centerline{\includegraphics[width=0.8\textwidth]{heatmap-Fst-1000G}}
\caption{Heatmap with clustering based on the $F_{ST}$ between pairs of the 26 1000G populations. Corresponding values are reported in table S1-S5. \label{fig:heatmap0}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.8\textwidth]{heatmap-bhattacharyya-1000G}}
	\caption{Heatmap with clustering based on the Bhattacharyya distances between pairs of the 26 1000G populations. \label{fig:heatmap1}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.8\textwidth]{heatmap-centers-1000G}}
	\caption{Heatmap with clustering based on the distances between centers of pairs of the 26 1000G populations. \label{fig:heatmap2}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.8\textwidth]{heatmap-closest-1000G}}
	\caption{Heatmap with clustering based on the shortest distances between pairs of the 26 1000G populations. \label{fig:heatmap3}}
\end{figure}

\FloatBarrier

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{PCA-based ancestry inference}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.95\textwidth]{proj-UKBB-to-1000G}}
	\caption{First 18 PC scores of the 1000G data (in black), onto which the UK Biobank data has been projected (in red). \label{fig:proj-UKBB}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.95\textwidth]{UKBB-mismatch}}
	\caption{PC scores (computed in the UK Biobank) colored by self-reported ancestry. On the left, these are 50,000 random individuals. On the right, these are the 47 individuals with some discrepancy between their self-reported-ancestry and or ancestry estimation. \label{fig:mismatch}}
\end{figure}

\input{../tables/matched-pop-KNN}
\input{../tables/infer-UKBB-pop}
\input{../tables/POPRES-matching}


\FloatBarrier

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{PCA-based ancestry grouping}

\input{../tables/ancestry-groups}


\begin{figure}[h]
\centerline{\includegraphics[width=0.8\textwidth]{hist-dist-GBR}}
\caption{ \label{fig:hist1}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.8\textwidth]{hist-dist-British}}
	\caption{ \label{fig:hist2}}
\end{figure}

\begin{figure}[h]
	\centerline{\includegraphics[width=0.8\textwidth]{hist-dist-overall-center}}
	\caption{ \label{fig:hist3}}
\end{figure}

\FloatBarrier

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%




\end{document}
